apps/ledger-live-desktop/src/newArch/WalletSync/hooks/useAddMember.ts

```diff
@@ -3,55 +3,71 @@ import {
   setMemberCredentials,
   setTrustchain,
 } from "@ledgerhq/trustchain/store";
-import { useMutation } from "@tanstack/react-query";
 import { useDispatch, useSelector } from "react-redux";
 import { setFlow } from "~/renderer/actions/walletSync";
 import { Flow, Step } from "~/renderer/reducers/walletSync";
 import { useTrustchainSdk, runWithDevice } from "./useTrustchainSdk";
-import { TrustchainResultType } from "@ledgerhq/trustchain/types";
+import { TrustchainResult, TrustchainResultType } from "@ledgerhq/trustchain/types";
+import { useEffect, useState } from "react";
 
 export function useAddMember({ device }: { device: Device | null }) {
   const dispatch = useDispatch();
   const sdk = useTrustchainSdk();
   const memberCredentials = useSelector(memberCredentialsSelector);
+  const [error, setError] = useState<Error | null>(null);
+
+  const [userDeviceInteraction, setUserDeviceInteraction] = useState(false);
+
+  useEffect(() => {
+    addMember();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
if we manage to avoid having to disable this checks, it's always better. but i understand it's not the easiest in here.
also small feedback: i think the addMember function can be inlined in the useEffect to reduce a bit the spaghetti & it can help seeing exactly what are the deps we need.

when a dep need to be static, a useRef can be done / if we just want to discard temporary case (eg if device is falsy) we can also discard the case (`if (!device) return`)